src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -1641,6 +1641,35 @@ public static bool IsOpen(this IEdmStructuredTypeReference type)
         }
 
         /// <summary>
+        /// Returns true if the definition of this reference is open.
+        /// </summary>
+        /// <param name="type">Reference to the calling object.</param>
+        /// <returns>If the definition of this reference is open.</returns>
+        public static bool IsOpen(this IEdmType type)
+        {
+            EdmUtil.CheckArgumentNull(type, "type");
+
+            IEdmStructuredType structuredType = type as IEdmStructuredType;
+            if (structuredType != null)
+            {
+                return structuredType.IsOpen;
+            }
+
+            // If its a collection, return whether its element type is open.
+            // This is because when processing a navigation property, the target type
+            // may be a collection type even though a key expression has been applied.
+            // This will be cleaned up in a subsequent change.
+            // TODO: when SingleResult is removed from the semantic path parser, change this to return false.

```

Comment:
Do we have a way to track this TODO? Should we open a GitHub issue and reference the number here?